recursive_diff/recursive_diff.py

```diff
@@ -318,15 +318,27 @@ def diff(msg, print_path=path):
                 diffs = (lhs.astype('<M8[ns]').astype(int)
                          != rhs.astype('<M8[ns]').astype(int))
 
+            elif lhs.dtype.kind != rhs.dtype.kind:
+                # The cases of comparison between different numeric types
+                # should have been captured in the blocks above.  As far as
+                # numpy's concern, values between diffeent non-numeric types
+                # are not compatible and therefore not equal.
+                diffs = numpy.full(lhs.shape, dtype=bool, fill_value=True)
+
```

Comment:
Not OK - if you try comparing, say, strings vs. dates you are supposed to get a single line stating a type mismatch; not one line of error per element in the array. This line is different from the old lines 328:329 below, which were converting a scalar "True" to a scalar array (``numpy.array(True)``)
Also missing a unit test.

Reply:
Did you notice that the outer block condition had been changed?  The original line 324 (diffs = lhs.values != rhs.values) is what caused the deprecation warning, which is what we want to move away from.   IMHO, we should move away from the original line 324.   Here my proposal is to check the dtypes of both LHS and RHS explicitly -- namely, if dtypes are different, the original line 324 would have returned a scaler, therefore we produce the scalar array -- by this way we achieve the same without involving the original line 324.